clang/include/clang/Basic/AttrDocs.td

```diff
@@ -3611,6 +3611,82 @@ where the last N*2 are as described above.
   }];
 }
 
+def SYCLAddIRAnnotationsMemberDocs : Documentation {
+  let Category = DocCatVariable;
+  let Heading = "add_ir_annotations_member";
+  let Content = [{
+This attribute can be applied to a non-static field. Access to a field with this
```

Comment:
I believe the initial use-case is for pointer types, but I don't see why we would need to limit it. [`clang::annotate`](https://godbolt.org/z/jjcMqW7PK) is okay with adding this sort of annotations on non-pointer fields, so this might as well be too.

Reply:
Can we add a test verifying that it works ok? (or just please point one to me in case I've missed it)

Reply:
Good idea. I have made the codegen tests run with both a pointer and a scalar.